sympy/parsing/c/c_parser.py

```diff
@@ -655,6 +657,180 @@ def transform_decl_stmt(self, node):
+                            [token.spelling,
+                            self.get_token_type(token)])
+
+                # process remaining operators
+                while (len(operators_stack) != 0):
+                    rhs = combined_variables_stack.pop()
+                    lhs = combined_variables_stack.pop()
+                    operator = operators_stack.pop()
+                    combined_variables_stack.append(
+                        self.perform_operation(lhs, rhs, operator))
+
+                # rhs value of the final assignment
+                if combined_variables_stack[-1][1] == 'identifier':
+                    rhs_value = self.get_expr_for_operand(
+                        combined_variables_stack[-1])
+                    return Assignment(Variable(lhs_value),
+                        Variable(rhs_value))
+                else:
+                    rhs_value = self.get_expr_for_operand(
+                        combined_variables_stack[-1])
+                    return Assignment(Variable(lhs_value), rhs_value)
+            else:
+                raise NotImplementedError(
+                    "Relational operator, bitwise operator and division \
+                    operator are not implemented yet!")
+
+        def priority_of(self, op):
+            """To get the priority of given operator"""
+            if op == '+' or op == '-':
+                return 1
+            if op == '*':
+                return 2
+            return 0
+
+        def perform_operation(self, lhs, rhs, op):
+            """Performs operation supported by the sympy core
+
+            Returns
+            =======
+
+            combined_variable: list
+                contains variable content and type of variable
+
+            """
+            lhs_value = self.get_expr_for_operand(lhs)
+            rhs_value = self.get_expr_for_operand(rhs)
+            if op == '+':
+                return [Add(lhs_value, rhs_value), 'expr']
+            if op == '-':
+                return [Add(lhs_value, -rhs_value), 'expr']
+            if op == '*':
+                return [Mul(lhs_value, rhs_value), 'expr']
+
+        def get_token_type(self, token):
+            """ Returns appropriate type of the token passed"""
+            if token.kind == cin.TokenKind.IDENTIFIER:
+                return 'identifier'
+            if token.kind == cin.TokenKind.LITERAL:
+                return 'literal'
+            if token.kind == cin.TokenKind.PUNCTUATION:
+                return 'punctuation'
+
+        def get_expr_for_operand(self, combined_variable):
+            """Returns SymPy Codegen AST node corresponding to
+            combined variable passed. Combined variable contains
+            variable content and type of variable
+            """
+            if combined_variable[1] == 'identifier':
+                return Symbol(combined_variable[0])
+            if combined_variable[1] == 'literal':
+                if '.' in combined_variable[0]:
```

Comment:
There must be a more robust way to check for float.

Reply:
Other method can be this:
<pre>
try:
    float(combined_variable[0])
    isFloat = True
except:
    isFloat = False
</pre>

Reply:
But, time complexity will be similar, I guess

Reply:
No way, this new method changed integer values too to float(though it was expected!) Although there is no way of knowing that given value was expected to be a float or integer, the only way seems that given literal has a decimal point in it (which we normally use in weakly typed languages such as python e.g. f=1.0).